### ISSUE TITLE
Shipping total hidden and Order Total wrong on Delivery page during Checkout

### ISSUE THREAD

**Issue by crystalized** [spree/spree#6443]
Currently, when you reach the Delivery page during Checkout, the first radio button shipping option is selected. That Shipping Total is not displayed in the Order Summary, but is still included in the Order Total.
At the very least, the shipping total should be displayed on the delivery page so the customer can see where the Order Total is coming from. Right now, the Order Total looks incorrect because it secretly includes the Shipping Total.
It would be awesome if the Shipping Total and Order Total would update dynamically in the Order Summary as the customer clicks on different radio button options. Or at least have an "Update" button on the Delivery page the same as you have an "Update" button for the Shopping Cart.

**Comment by JDutil** [comment]
Please submit a PR if you could.

**Comment by crystalized** [comment]
I created a rough version of this by copying the strategy used to update the shopping cart.

I made the Shipping Total permanently visible in the Order Summary on all checkout pages. Then I copied the Update button code from the shopping cart's page to the _delivery.html.erb page.
Then I slightly modified the update method in the checkout_controller.rb:

        if @order.state == "delivery" && params.has_key?(:update)
        	redirect_to checkout_state_path(@order.state) and return
        end

This allows the customer to click the Update button on the shipping selection page to update the shipping costs without being moved to the next page for payment options. The "save and continue" button is still there and what they can use to move forward in the checkout process.
